yarn-project/noir-protocol-circuits/src/crates/types/src/abis/nullifier_key_validation_request.nr

```diff
@@ -0,0 +1,109 @@
+use dep::std::cmp::Eq;
+use crate::{
+    address::AztecAddress,
+    traits::Empty,
+    grumpkin_point::GrumpkinPoint,
+    grumpkin_private_key::GrumpkinPrivateKey,
+};
+
+global NULLIFIER_KEY_VALIDATION_REQUEST_SERIALIZED_LEN = 4;
+global NULLIFIER_KEY_VALIDATION_REQUEST_CONTEXT_SERIALIZED_LEN = 5;
+
+struct NullifierKeyValidationRequest {
+    public_key: GrumpkinPoint,
+    secret_key: GrumpkinPrivateKey,
+}
+
+impl Eq for NullifierKeyValidationRequest {
+    fn eq(self, request: NullifierKeyValidationRequest) -> bool {
+        (request.public_key.eq(self.public_key))
+            & (request.secret_key.eq(self.secret_key))
+    }
+}
+
+impl Empty for NullifierKeyValidationRequest {
+    fn empty() -> Self {
+        NullifierKeyValidationRequest {
+            public_key: GrumpkinPoint::zero(),
+            secret_key: GrumpkinPrivateKey::zero(),
+        }
+    }
+}
+
+impl NullifierKeyValidationRequest {
+    pub fn is_empty(self) -> bool {
+        self.public_key.is_zero() & self.secret_key.is_zero()
+    }
```

Comment:
You don't need to implement is_empty, there is a method in types::traits that tells you if any struct that implements Eq + Empty is actually empty :D

Reply:
Ohh that's nice!